java/src/main/native/src/row_conversion.cu

```diff
@@ -2110,33 +2172,57 @@ std::unique_ptr<table> convert_from_rows(lists_column_view const &input,
 
   int shmem_limit_per_tile = total_shmem_in_bytes / NUM_TILES_PER_KERNEL_LOADED;
 
-  auto iter = thrust::make_transform_iterator(thrust::make_counting_iterator(0), [&schema](auto i) {
-    return std::make_tuple(schema[i], nullptr);
-  });
-  auto column_info = detail::compute_column_information(iter, iter + num_columns);
-  auto const fixed_width_size_per_row =
-      util::round_up_unsafe(column_info.fixed_width_size_per_row, JCUDF_ROW_ALIGNMENT);
+  auto column_info = detail::compute_column_information(string_schema.begin(), string_schema.end());
+  auto const size_per_row = util::round_up_unsafe(column_info.size_per_row, JCUDF_ROW_ALIGNMENT);
 
   // Ideally we would check that the offsets are all the same, etc. but for now
   // this is probably fine
-  CUDF_EXPECTS(fixed_width_size_per_row * num_rows == child.size(),
-               "The layout of the data appears to be off");
-  auto dev_col_starts = make_device_uvector_async(column_info.fixed_width_column_starts, stream);
-  auto dev_col_sizes = make_device_uvector_async(column_info.fixed_width_column_sizes, stream);
+  CUDF_EXPECTS(size_per_row * num_rows <= child.size(), "The layout of the data appears to be off");
+  auto dev_col_starts = make_device_uvector_async(column_info.column_starts, stream);
+  auto dev_col_sizes = make_device_uvector_async(column_info.column_sizes, stream);
 
   // Allocate the columns we are going to write into
   std::vector<std::unique_ptr<column>> output_columns;
+  std::vector<std::unique_ptr<column>> string_row_offset_columns;
+  std::vector<std::unique_ptr<column>> string_length_columns;
   std::vector<int8_t *> output_data;
   std::vector<bitmask_type *> output_nm;
-  for (int i = 0; i < static_cast<int>(num_columns); i++) {
-    auto column =
-        make_fixed_width_column(schema[i], num_rows, mask_state::UNINITIALIZED, stream, mr);
-    auto mut = column->mutable_view();
-    output_data.emplace_back(mut.data<int8_t>());
-    output_nm.emplace_back(mut.null_mask());
-    output_columns.emplace_back(std::move(column));
+  std::vector<int32_t *> string_row_offsets;
+  std::vector<int32_t *> string_lengths;
+  for (auto i : schema) {
```

Comment:
I thought about doing that, but it's a transform with side-effects because we're actually building 5 vectors in this loop. I felt it more obvious that multiple things were happening to use a for loop instead of a transform that took some reference variables and messed with them as well.